File: server/router/kubernetes/services.go
package kubernetes

import (
	"github.com/gin-gonic/gin"
	v1 "github.com/pddzl/kubeops/server/api/v1"
	"github.com/pddzl/kubeops/server/middleware"
)

type ServicesRouter struct{}

func (s *ServicesRouter) InitServicesRouter(Router *gin.RouterGroup) {
	servicesRouter := Router.Group("services").Use(middleware.OperationRecord())
	servicesRouterWithoutRecord := Router.Group("services")
	servicesApi := v1.ApiGroupApp.KubernetesApiGroup.ServicesApi
	{
		servicesRouter.POST("getServicesDetail", servicesApi.GetServicesDetail) // 获取services详情
		servicesRouter.POST("getServicesRaw", servicesApi.GetServicesRaw)       // 获取services in raw
		servicesRouter.POST("getServicesPods", servicesApi.GetServicesPods)     // 获取services pods
	}
	{
		servicesRouterWithoutRecord.POST("getServicesList", servicesApi.GetServicesList) // 获取所有services
	}
}

File: web/src/api/kubernetes/services.js
import service from '@/utils/request'

// 获取services list
export const getServicesList = (data) => {
  return service({
    url: '/services/getServicesList',
    method: 'post',
    data
  })
}

// 获取services in raw
export const getServicesRaw = (data) => {
  return service({
    url: '/services/getServicesRaw',
    method: 'post',
    data
  })
}

// 获取services in detail
export const getServicesDetail = (data) => {
  return service({
    url: '/services/getServicesDetail',
    method: 'post',
    data
  })
}

// 获取services关联pods
export const getServicesPods = (data) => {
  return service({
    url: '/services/getServicesPods',
    method: 'post',
    data
  })
}

// 删除services
export const deleteServices = (data) => {
  return service({
    url: '/services/deleteServices',
    method: 'post',
    data
  })
}
